staging/src/k8s.io/cli-runtime/pkg/resource/builder.go

```diff
@@ -768,7 +795,13 @@ func (b *Builder) mappingFor(resourceOrKindArg string) (*meta.RESTMapping, error
 		gvk, _ = restMapper.KindFor(*fullySpecifiedGVR)
 	}
 	if gvk.Empty() {
-		gvk, _ = restMapper.KindFor(groupResource.WithVersion(""))
+		gvks, _ := restMapper.KindsFor(groupResource.WithVersion(""))
```

Comment:
Although this PR looks promising, unfortunately `KindsFor` does not return the list of resources belonging to the given short name. Because `KindsFor` calls `expandResourceShortcut` in https://github.com/kubernetes/kubernetes/blob/9924dc65b75975c5993da95ea8e0cb9360543c3f/staging/src/k8s.io/client-go/restmapper/shortcut.go#L62 and this function https://github.com/kubernetes/kubernetes/blob/9924dc65b75975c5993da95ea8e0cb9360543c3f/staging/src/k8s.io/client-go/restmapper/shortcut.go#L155 already picks the one of the resources instead the list of resources with given short name.